### PR TITLE
Devdocs 2640 update webhooks admin api documentation for open beta

### PR DESCRIPTION
**Pull request by markcmurphy**
[DEVDOCS-2640](https://jira.bigcommerce.com/browse/DEVDOCS-2640)

This involved merging the `api-specs` version of `webhooks.v3.yml` file with the version containing the beta Webhooks Admin endpoint. To compare them item by item, I changed the order of objects to match the OAS specs which accounts for most of the additions and deletions. I combined info or selected whichever was most comprehensive. For instance, the beta Webhooks Admin doc had some error codes for POST `/webhooks` that aren't in the `api-specs` repo.

I published a preview version that might help with reviewing:   

[Stoplight Front-end Preview](https://bc4wp1.docs.stoplight.io/api-reference/storefront/webhooks2)

### CODE REVIEW

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:5]:
```suggestion
    Create and manage webhooks to get notified when specific events occur on stores. For more information, see [Webhooks Overview](https://developer.bigcommerce.com/api-docs/store-management/webhooks/overview).

```

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:10]:
```suggestion

```

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:25]: Can we have two serves referenced on the same page? I believe we should separate V2 and V3 APIs.

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:41]:
```suggestion
```

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:44]: Do we need to mention the V2 API in this documentation?

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:102]:
```suggestion
      description: Creates a webhook. You can only create one webhook at a time. Custom headers are permitted. Destination URL must be served on port 443 (custom ports are not currently supported).

```

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:228]:
```suggestion
          description: Webhook with specified ID doesn't exist.

```

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:244]:
```suggestion
      description: 'Deletes a webhook. You can create only one webhook at a time. When a webhook is deleted, it is returned in the response as a `200` OK.'

```

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:289]:
```suggestion

```

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:292]: Do we need to mention the V2 REST API in this documentation?

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:528]: I see you are changing the naming convention. Is there a reason for this update?

File: reference/webhooks.v3.yml
Comment (ektravel) [inline reference/webhooks.v3.yml:540]:
```suggestion
          description: Headers to be included in destination URL request.

```

**Review by ektravel — COMMENTED** [review]
I don't think we should be referencing the V2 REST API unless absolutely necessary. Similarly, I don't think we should list both servers (V2 and V3). When it comes to changing the naming convention, I will defer to @uwikaiddi.

File: reference/webhooks.v3.yml
Comment (markcmurphy) [reply reference/webhooks.v3.yml:43]: I left the mention since it was in the API beta docs. Let's see what @uwikaiddi says about whether we should break this spec file into two menu items for `V2` and `V3` or leave consolidated, in which case we should probably point out that both versions are available.

File: reference/webhooks.v3.yml
Comment (markcmurphy) [reply reference/webhooks.v3.yml:293]: Same as above about waiting to see if we are going to break this into two spec files for `V2` and `V3`.

**Review by uwikaiddi — APPROVED** [review]

**Comment by markcmurphy** [comment]
Converting to draft because I have paused DEVDOCS-2640 until I receive a go-ahead from webhooks.

File: reference/webhooks.v3.yml
Comment (jyuricle) [inline reference/webhooks.v3.yml:13]: We should take into account that V2 & V3 endpoints are a bit different. For example, `getAllWebhooks` endpoint for v2 doesn't support pagination. Also, in v2 response json is not wrapped in `data` parent field. So it might have sense to differentiate it until v2 is not fully removed.

File: reference/webhooks.v3.yml
Comment (jyuricle) [inline reference/webhooks.v3.yml:528]: On the UI it shows that these fields have 3 validations and the interval for them is from -9223372... to 9223372. At least the minimum should be 0.

File: reference/webhooks.v3.yml
Comment (jyuricle) [inline reference/webhooks.v3.yml:87]: V3 pagination which is used for webhooks as well has different structure: 
```suggestion
                      pagination:
                        total: 0
                        count: 0
                        per_page: 0
                        current_page: 0
                        total_pages: 0
                        links:
                          previous: ?limit=0&page=0
                          current: ?limit=0&page=0
                          next: ?limit=0&page=0

```

File: reference/webhooks.v3.yml
Comment (jyuricle) [inline reference/webhooks.v3.yml:555]: Actually it is not required, but `is_active=true` by default

File: reference/webhooks.v3.yml
Comment (jyuricle) [inline reference/webhooks.v3.yml:167]: This schema misses such fields as `client_id`, `created_at`, `updated_at`, `store_hash`. It should be the same webhook schema as in `getAllWebhooks`

File: reference/webhooks.v3.yml
Comment (jyuricle) [inline reference/webhooks.v3.yml:218]: The same as in `Get webhook`, schema is not complete

File: reference/webhooks.v3.yml
Comment (jyuricle) [inline reference/webhooks.v3.yml:257]: The same as in `Get webhook`, schema is not complete

File: reference/webhooks.v3.yml
Comment (jyuricle) [inline reference/webhooks.v3.yml:326]: This is represented in JSON a bit differently: 
```
    "blocked_domains": [
      {
        "destination": "string",
        "time_left": 0,
        "reasons": [
          {
            "failure_description": "string",
            "count": 0,
            "timestamp": 0
          }
        ]
      }
 ```

File: reference/webhooks.v3.yml
Comment (jyuricle) [inline reference/webhooks.v3.yml:428]: Should 204 with `No content`

**Pull request closed without merging**
